packages/mobile/src/screens/track-screen/TrackScreenDownloadButtons.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback } from 'react'
+
+import {
+  Name,
+  ButtonState,
+  useDownloadTrackButtons,
+  tracksSocialActions
+} from '@audius/common'
+import type { ID, ButtonType as DownloadButtonType } from '@audius/common'
+import { View } from 'react-native'
+import { useDispatch } from 'react-redux'
+
+import IconDownload from 'app/assets/images/iconDownload.svg'
+import { Button } from 'app/components/core'
+import LoadingSpinner from 'app/components/loading-spinner'
+import { useToast } from 'app/hooks/useToast'
+import { make, track } from 'app/services/analytics'
+import { makeStyles } from 'app/styles'
+const { downloadTrack } = tracksSocialActions
+
+export type DownloadButtonProps = {
+  state: ButtonState
+  type: DownloadButtonType
+  label: string
+  hasDownloadAccess: boolean
+  onClick?: () => void
+}
+
+export const messages = {
+  followToDownload: 'Must follow artist to download',
+  addDownloadPrefix: (label: string) => `Download ${label}`,
+  mustHaveAccess: 'Must have access to download'
+}
+
+const useStyles = makeStyles(() => ({
+  buttonContainer: {
+    alignSelf: 'center',
+    marginBottom: 6
+  }
+}))
+
+const DownloadButton = ({
+  label,
+  state,
+  hasDownloadAccess,
+  onClick = () => {}
+}: DownloadButtonProps) => {
+  const { toast } = useToast()
+
+  const styles = useStyles()
+  const requiresFollow = state === ButtonState.REQUIRES_FOLLOW
+  const isProcessing = state === ButtonState.PROCESSING
+  const isDisabled =
```

Comment:
should there be a `does_current_user_follow` flag check along with the `requiresFollow`?

Reply:
just tested, it already checks that the current user follows the artist upstream! good callout